lnd-rpc/src/main/scala/org/bitcoins/lnd/rpc/LndRpcClient.scala

```diff
@@ -449,6 +448,58 @@ class LndRpcClient(val instance: LndInstance, binary: Option[File] = None)(
       }
   }
 
+  def getTransaction(txId: DoubleSha256DigestBE): Future[Option[TxDetails]] = {
+    // Idk why they don't have a separate function to just get one tx
+    getTransactions.map(_.find(_.txId == txId))
+  }
+
+  def getTransactions: Future[Vector[TxDetails]] = {
+    getTransactions(GetTransactionsRequest())
+  }
+
+  def getTransactions(startHeight: Int): Future[Vector[TxDetails]] = {
+    getTransactions(startHeight, -1)
+  }
+
+  def getTransactions(
+      startHeight: Int,
+      endHeight: Int): Future[Vector[TxDetails]] = {
+    getTransactions(GetTransactionsRequest(startHeight, endHeight))
+  }
+
+  def getTransactions(
```

Comment:
unit test? This seems like a complex method

Reply:
https://github.com/bitcoin-s/bitcoin-s/pull/2959/files#diff-810ac1d1deaae1d8a031f3ffaf6c02f17287473193f17f248efdbcd3df82140dR72